### PR TITLE
pydrive2: init at 1.10.0

### PR DESCRIPTION
**Pull request by StephenWithPH**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/97268#issuecomment-687736016

###### Things done
~Still a WIP... still wrapping my head around Python and nix.~

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -9574,4 +9574,10 @@
       fingerprint = "F1C5 760E 45B9 9A44 72E9  6BFB D65C 9AFB 4C22 4DA3";
     }];
   };
+  stephenwithph = {
+    name = "StephenWithPH";
+    email = "StephenWithPH@users.noreply.github.com";
+    github = "StephenWithPH";
+    githubId = 2990492;
+  };
 }
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:9583]: Please make this into a separate commit:

```
maintainers: add <name>
```

also, please attempt to sort your entry

File: pkgs/development/python-modules/pydrive2/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ lib, python3 }:
+
+python3.pkgs.buildPythonPackage rec {
```
Comment (jonringer) [inline pkgs/development/python-modules/pydrive2/default.nix:3]: you shouldn't import a specific python version please look at https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies

**Review by jonringer — COMMENTED** [review]
If you're new to nix python packaging, please take a look https://www.youtube.com/watch?v=jXd-hkP4xnU where I cover python packaging within nixpkgs. Also, https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies is a good resource, although it's a long read.

File: pkgs/development/python-modules/pydrive2/default.nix (hunk 2 above)
Comment (StephenWithPH) [reply pkgs/development/python-modules/pydrive2/default.nix:3]: Before trying to handle Python 2 vs 3, I received the error: `google-api-core-1.22.1 not supported for interpreter python2.7.` I'm reasonably confident this comes from...

https://github.com/NixOS/nixpkgs/blob/33402a812d5c0addb206d15e1426df007eea8b73/pkgs/development/python-modules/google_api_core/default.nix#L9

... transitively due to `google_api_python_client`.

I was unable to get incantations of `disabled = ! isPy3k` or `disabled = isPy27` to work... builds still gave me a Python 2 interpreter error. I ended up with a pattern that worked by cribbing from other PRs.

What am I missing?

Comment (StephenWithPH) [reply pkgs/development/python-modules/pydrive2/default.nix:3]: @jonringer ... thoughts?

Comment (jonringer) [reply pkgs/development/python-modules/pydrive2/default.nix:3]: Using `pythonXPackages` isn't allowed from `python-modules` as this will mix interpreter versions for a given package. For example, if you were to do `python37Packages.<mypkg>`, but the expression pulled from `python3[8]Packages`. Python will be mis-importing several libraries, and this will likely cause issues at runtime.

Please refer to [Python dependency handling](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies) on the best practices for handling dependencies within `python-modules`. Namely, you should be importing each python package individually, and not refer to another python package set (E.g. `python3Packages`).

If you're new to nixpkgs, you're free to view some of my videos on nixpkgs and python:
- [Python packaging using nixpkgs](https://www.youtube.com/watch?v=jXd-hkP4xnU)
- [Adding a package to nixpkgs unstable](https://www.youtube.com/watch?v=fvj8H5yUKu8)

Comment (jonringer) [reply pkgs/development/python-modules/pydrive2/default.nix:3]: convert it over to using the individual packages and have it build for python3, and we'll go from there :)

File: pkgs/development/python-modules/pydrive2/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  src = python3.pkgs.fetchPypi {
+    pname = "PyDrive2";
+    inherit version;
+    sha256 = "0b803dq6g13zypqplwi4bzdjmsdrxmvfiba9121wma51j83350kp";
+  };
+
+  doCheck = false; # the tests require your GCP credentials
+
+  propagatedBuildInputs = with python3.pkgs; [
+    google_api_python_client
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pydrive2/default.nix:16]:
```suggestion
    google-api-python-client
```

File: pkgs/development/python-modules/pydrive2/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    oauth2client
+    pyopenssl
+    pyyaml
+  ];
+
+  meta = with lib; {
+    description = "Google Drive API Python wrapper library";
+    homepage = "https://github.com/iterative/PyDrive2";
+    license = licenses.asl20;
+    maintainers = [ stephenwithph ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pydrive2/default.nix:26]: Please add yourself as a maintainer with the message ``maintainers: add stephenwithph``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pydrive2/default.nix (hunk 4 above)
Comment (StephenWithPH) [reply pkgs/development/python-modules/pydrive2/default.nix:36]: I made it on the list via a different package contribution: https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix#L11149

File: pkgs/development/python-modules/pydrive2/default.nix (hunk 3 above)
Comment (StephenWithPH) [reply pkgs/development/python-modules/pydrive2/default.nix]: fixed in 5b86966d9a6cbb4d35419cd370d467727887bc40

File: pkgs/development/python-modules/pydrive2/default.nix (hunk 2 above)
Comment (StephenWithPH) [reply pkgs/development/python-modules/pydrive2/default.nix]: I picked this back up. I think I did all the needful in 5b86966d9a6cbb4d35419cd370d467727887bc40

File: pkgs/development/python-modules/pydrive2/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{ lib,
+  python3,
+  fetchPypi,
+  google-api-python-client,
+  oauth2client,
+  pyopenssl,
+  pyyaml,
+  six,
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pydrive2/default.nix:8]:
```suggestion
{ lib,
, python
, fetchPypi
, google-api-python-client
, oauth2client
, pyopenssl
, pyyaml
, six
```

File: pkgs/development/python-modules/pydrive2/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ }:
+
+python3.pkgs.buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pydrive2/default.nix:11]:
```suggestion
buildPythonPackage rec {
```

File: pkgs/development/python-modules/pydrive2/default.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  version = "1.10.0";
+
+  src = fetchPypi {
+    pname = "PyDrive2";
+    inherit version;
+    sha256 = "1av3fkbq8xvrvshy8yiq6ysllwll1mfcaspv4aww1xhyzys1kggp";
+  };
+
+  # the tests require GCP credentials
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pydrive2/default.nix:22]: Please add a pythonImportsCheck instead and move doCheck after all *buildInputs.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pydrive2/default.nix (hunk 5 above)
Comment (StephenWithPH) [reply pkgs/development/python-modules/pydrive2/default.nix]: fixed in decb45313390f19369ceeea3d34700add2c10700

File: pkgs/development/python-modules/pydrive2/default.nix (hunk 6 above)
Comment (StephenWithPH) [reply pkgs/development/python-modules/pydrive2/default.nix]: fixed in decb45313390f19369ceeea3d34700add2c10700

File: pkgs/development/python-modules/pydrive2/default.nix (hunk 7 above)
Comment (StephenWithPH) [reply pkgs/development/python-modules/pydrive2/default.nix]: fixed in decb45313390f19369ceeea3d34700add2c10700

**Comment by SuperSandro2000** [comment]
Closing in favor of #156195
